### ISSUE TITLE
[Translation] Add generic extraction marker function

### ISSUE THREAD

**Issue by ro0NL** [symfony/symfony#35418]
Extracting translation keys from source is nice, and saves us a few steps and makes thing less error prone in general.

Currently, in PHP, only `TranslatorInterface::trans()` in controllers calls are extracted as such. However, often we have just a translatable string, which is passed to the translator later on. E.g. with knp menus:

```php
$menu->addChild('translation.key', ['extras' => ['translation_domain' => 'nav']]);
```

To extract `translation.key` i propose to add `$string = Translatable::_('translation.key')`. Which is an identity function defined in contracts. It enables extractors to act upon. More short, maybe `t('string')` even ...

Next, we should scrape entire `src` IMHO, or any source file from the DI container.

Related, can we consider https://github.com/php-translation/extractor for core? Or why is there something competing doing a better job?

**Comment by ro0NL** [comment]
Actually `_()` could return a `Translatable` value object, so can even pass the params + domain;

```php
$t = Translatable:_('extracted {param}', ['{param}' => 'replacement'], 'my-domain')
$t->getDomain();
$t->getParameters();
(string) $t; // "extracted {param}"

echo Translatable::trans($someTranslator, $t); // "extracted replacement"
```

**Comment by ro0NL** [comment]
fixed by #37670
